### ISSUE TITLE
Multiple XHR requests on page visit are not always correctly detected and awaitable

### ISSUE THREAD

**Issue by ambvdijk** [cypress-io/cypress#7361]
### Current behavior:

Multiple XHR requests on page visit are not always correctly detected and awaitable.

![image](https://user-images.githubusercontent.com/8982010/82043499-fc054800-96ab-11ea-8ced-f8a65561f5d0.png)

This step will fail 60% of the tests runs.
It seems the faster the server responds, the more it fails.

### Desired behavior:

All XHR request on page visit can be awaited in a stable fashion.

### Test code to reproduce

```javascript
  beforeEach(() => {
    cy.server();
    cy.route('/application/load/chart/*').as('chartUpdate');

    cy.visit('/application/page');

    cy.wait(['@chartUpdate', '@chartUpdate', '@chartUpdate', '@chartUpdate', '@chartUpdate']);
  });
```
### Versions

Cypress: 4.3.0
Operating System: Windows 10
Chrome: 81
Electron: 80

**Comment by ambvdijk** [comment]
This might be similar (or the same as) issue #5999

**Comment by ambvdijk** [comment]
Closed as being a duplicate of #5999
